### PR TITLE
Add a --tms-user-agent option

### PR DESCRIPTION
**Pull request by MrTrustor**
This option lets the user customize the User Agent
that gpx animator uses when making requests to a TMS server.

Some TMS servers enforce rules on which user agents can be used.

I'm not sure if you're interested in this feature, but I needed it for my own usage.

The german help message comes from Google Translate, so not sure how good it is.

### CODE REVIEW

**Comment by McPringle** [comment]
Thank you very much for your contribution, this can be a very useful feature. I'll do the review and merge ASAP.

File: src/main/java/app/gpx_animator/core/renderer/cache/TileCache.java [hunk 1]
```diff
@@ -78,32 +78,36 @@ public static void ageCache() {
...
     }
 
-    private static BufferedImage unCachedGetTile(final String url) throws UserException {
+    private static BufferedImage unCachedGetTile(final String url, final String userAgent) throws UserException {
         BufferedImage mapTile;
 
-        final var userAgent = String.format("%s %s on %s %s (%s)", //NON-NLS
-                Constants.APPNAME, Constants.VERSION, Constants.OS_NAME, Constants.OS_VERSION, Constants.OS_ARCH);
-        System.setProperty("http.agent", userAgent);
+        if (userAgent != null) {
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/renderer/cache/TileCache.java:101]: In `Configuration.java` you set the user agent to an empty string. To be safe, you should do a `userAgent.isBlank()` check here, too.

**Review by McPringle — CHANGES_REQUESTED** [review]
I have found a very small issue, easy to fix. In addition, this setting should be added to the GUI. If you are unsure, how to do that, I can do that in the next few days.

File: src/main/java/app/gpx_animator/core/renderer/cache/TileCache.java (hunk 1 above)
Comment (MrTrustor) [reply src/main/java/app/gpx_animator/core/renderer/cache/TileCache.java]: Fixed

**Comment by MrTrustor** [comment]
Thanks a lot, I've fixed the issue.
For the GUI, sorry, I don't know how to do that... It's the first time I write any code in Java in 10 years I think!

**Review by McPringle — APPROVED** [review]
Thank you very much for the fast change!

**Comment by McPringle** [comment]
Nice PR, thank you very much for your work. I added your GitHub profile to the list of contributors and the user agent option to the UI. 👍

**Pull request merged**